src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js

```diff
@@ -27,42 +26,9 @@ const buildSearchOption = (id, name, searchGroups, searchScope, type) => ({
         })),
 });
 
-const useTrackedEntityTypesWithCorrelatedPrograms = (): TrackedEntityTypesWithCorrelatedPrograms =>
-    useMemo(() =>
-        [...programCollection.values()]
-            .filter(program => program instanceof TrackerProgram)
-            // $FlowFixMe
-            .reduce((acc, {
-                id: programId,
-                name: programName,
-                trackedEntityType: {
-                    id: trackedEntityTypeId,
-                    name: trackedEntityTypeName,
-                    searchGroups: trackedEntityTypeSearchGroups,
-                },
-                searchGroups,
-            }: TrackerProgram) => {
-                const accumulatedProgramsOfTrackedEntityType =
-            acc[trackedEntityTypeId] ? acc[trackedEntityTypeId].programs : [];
-                return {
-                    ...acc,
-                    [trackedEntityTypeId]: {
-                        trackedEntityTypeId,
-                        trackedEntityTypeName,
-                        trackedEntityTypeSearchGroups,
-                        programs: [
-                            ...accumulatedProgramsOfTrackedEntityType,
-                            { programId, programName, searchGroups },
-                        ],
-
-                    },
-                };
-            }, {}),
-    [],
-    );
-
-const useSearchOptions = (trackedEntityTypesWithCorrelatedPrograms): AvailableSearchOptions =>
-    useMemo(() =>
+const useSearchOptions = (): AvailableSearchOptions => {
+    const trackedEntityTypesWithCorrelatedPrograms = useTrackedEntityTypesWithCorrelatedPrograms();
```

Comment:
I guess I realised how flexible hooks are and that I didnt really have to pass it as a argument in many cases. I change that in this PR. Take a look and let me know if you see anything strange